pkg/kubelet/leaky/leaky.go

```diff
@@ -1,25 +0,0 @@
-/*
-Copyright 2015 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-// Package leaky holds bits of kubelet that should be internal but have leaked
-// out through bad abstractions.  TODO: delete all of this.
-package leaky
-
-const (
-	// PodInfraContainerName is used in a few places outside of Kubelet, such as indexing
-	// into the container info.
-	PodInfraContainerName = "POD"
```

Comment:
FWIW: CRI-O uses this field as well. I am open to moving it, but I think it should be exported. Probably also worth a comment that's why it's exported

Reply:
Thanks for your feedback @haircommander 
According to the package comment, it seems that this field should not be exposed.
But if this field is already used by other projects, maybe we should move it to a more reasonable location, such as `k8s.io/kubelet`. 

What do you think? @SergeyKanzhelev @bobbypage

Reply:
The unfortunate part of this is that it already leaked out of the codebase - and so just removing it isn't something we can reasonably do. 

Removing the package doesn't really fix the abstraction, but removes the bandage :(

Reply:
Agree, we need to evaluate how to deal with the leaked field again.